### PR TITLE
Fix directory transfer error

### PR DESCRIPTION
**Pull request by sturoscy-personal**
Just going to get this in for review. Needs a bunch of testing :) 

Hopefully will mostly take care of:
https://app.shortcut.com/globus/story/15821
https://app.shortcut.com/globus/story/15820
https://app.shortcut.com/globus/story/15819
https://app.shortcut.com/globus/story/15859

### CODE REVIEW

File: src/components/Endpoint.tsx [hunk 1]
```diff
@@ -99,21 +109,44 @@ const Endpoint = (props) => {
...
     } catch (error) {
+      setAPIError({ ...error, ...{ global: true } });
       let error_response = await error.response.json();
-      /* Note: This probably isn't a great UX to simply pop up a login page, but it
-      does demonstrate the base functionality for picking endpoints */
+
+      /* 
+        Note: This probably isn't a great UX to simply pop up a login page, but it
+        does demonstrate the base functionality for picking endpoints 
+      */
```
Comment (NickolausDS) [inline src/components/Endpoint.tsx:126]: We should put in a ticket for updating this behavior.

File: src/components/Endpoint.tsx [hunk 2]
```diff
@@ -181,35 +214,39 @@ const Endpoint = (props) => {
...
-      destination_endpoint: destinationEndpoint,
-      DATA: transferItems,
-    };
+    } else {
+      // Loop through selectedEndpointItems from state
+      for (let selectedEndpointItem of selectedEndpointItems) {
+        transferItems.push({
+          source_path: `${endpointList.path}${selectedEndpointItem.name}`,
+          destination_path: `${config.collection_base_path}/${props.selectedJupyterItems.directories[0].path}/${selectedEndpointItem.name}`,
+          recursive: selectedEndpointItem.type == 'dir' ? true : false,
```
Comment (NickolausDS) [inline src/components/Endpoint.tsx:228]: I see destination has a path separator between items, but source doesn't. Does it need one?

File: src/components/Endpoint.tsx (hunk 1 above)
Comment (sturoscy-personal) [reply src/components/Endpoint.tsx:126]: Sounds good. I'll need to brainstorm with you how you might want to change the behavior.

File: src/components/Endpoint.tsx [hunk 3]
```diff
@@ -181,35 +214,39 @@ const Endpoint = (props) => {
     if (props.selectedJupyterItems.directories.length === 0 || props.selectedJupyterItems.directories.length > 1) {
       setLoading(false);
       setAPIError({
-        response: { status: '500', statusText: 'Please select one jupyter directory to transfer data to' },
+        response: {
+          status: '500',
+          statusText: 'To transfer to Jupyter Hub, you must select only one directory to transfer to.',
```
[the remark this answers is not in the document]
Comment (sturoscy-personal) [reply src/components/Endpoint.tsx:219]: `apiError` component is looking for a `status` key https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L321 Happy to work this out differently.

File: src/components/Endpoint.tsx (hunk 2 above)
Comment (sturoscy-personal) [reply src/components/Endpoint.tsx:228]: It should not need one. Did you get any errors?

**Comment by NickolausDS** [comment]
Feel free to add a front-end linter to the pre-commit checker. The one for `black` is actually working quite well!

File: src/components/Endpoint.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (NickolausDS) [reply src/components/Endpoint.tsx:219]: Is `status` only used for checking the types of errors that come from JupyterLab? The numerical codes make me think of http server responses, where 500 would be an unhandled failure in the service. String codes like `DirectorySelectionError` may work better here for a re-usable set of responses that you could check programatically.

File: src/components/Endpoint.tsx (hunk 2 above)
Comment (sturoscy-personal) [reply src/components/Endpoint.tsx:228]: I could add some logic to check if a path separator is needed or not.

Comment (sturoscy-personal) [reply src/components/Endpoint.tsx:238]: https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L228-L229
https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L284-L288
https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/src/components/Endpoint.tsx#L300-L304

**Comment by sturoscy-personal** [comment]
> Feel free to add a front-end linter to the pre-commit checker. The one for `black` is actually working quite well!



> Feel free to add a front-end linter to the pre-commit checker. The one for `black` is actually working quite well!

https://github.com/globus/globus-jupyterlab/blob/fix-directory-transfer-error/.pre-commit-config.yaml#L12

File: src/components/Endpoint.tsx (hunk 1 above)
Comment (NickolausDS) [reply src/components/Endpoint.tsx:129]: Added a ticket for this: https://app.shortcut.com/globus/story/16122

**Comment by NickolausDS** [comment]
Looks good!

**Pull request merged**